### PR TITLE
CI: use pipx and build to simplify and remove a workaround

### PR DESCRIPTION
**Pull request by henryiii**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

This cleans up the SDist build slightly. First, it uses PyPA's "build" instead of directly calling `setup.py`, which will respect PEP 518 requirements and avoids the workaround of installing NumPy to be able to run setup.py. It also uses pipx to avoid needless setup; PyPA's pipx is provides as a first-class package manager on GitHub Actions (and Azure, they share images).

### CODE REVIEW

File: .github/workflows/sdist.yml [hunk 1]
```diff
@@ -38,17 +38,9 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Install dependencies
-      run: |
```
Comment (jreback) [inline .github/workflows/sdist.yml:42]: the idea is to simulate core building and distributing this application. how does this improve the experience here?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
> PyPA's pipx is provides as a first-class package manager on GitHub Actions (and Azure, they share images).

this doesn't help as the archives are not generated here (they are actually justed tested here). sure i suppose we could move to this workflow, but that should happen in conjuction with a change like this.

File: .github/workflows/sdist.yml (hunk 1 above)
Comment (henryiii) [reply .github/workflows/sdist.yml:42]: Artifact building should happen with `build`, as well. `build` will respect PEP 518 requirements, raw `setup.py` commands cannot, which is why the workaround in #39416 was introduced. Making an SDist should be done with `pip install build && python -m build --dist` (or `pipx run build`), not `pip install setuptools wheel numpy && python setup.py sdist --formats=gztar`.

**Comment by henryiii** [comment]
Where are the SDist's built? https://github.com/MacPython/pandas-wheels only does wheels, AFAICT.

Also, this is still testing that the artifact builds and is installable, it's just using better infrastructure for doing so without manual workarounds in producing them.

File: .github/workflows/sdist.yml (hunk 1 above)
Comment (jreback) [reply .github/workflows/sdist.yml:42]: can you replace with `pip install build && python -m build --dist` prefer to use the standard tools here

File: .github/workflows/sdist.yml [hunk 2]
```diff
@@ -38,17 +38,9 @@ jobs:
...
-        python -m pip install --upgrade pip setuptools wheel
-
-        # GH 39416
-        pip install numpy
-
     - name: Build pandas sdist
       run: |
-        pip list
```
Comment (jbrockmendel) [inline .github/workflows/sdist.yml:50]: i think we usually want this line so that we can see what versions are installed when debugging CI failures

Comment (henryiii) [reply .github/workflows/sdist.yml:50]: This is not the build environment. No version listed here would have anything to do with what you have inside the PEP 517 build environment. If you use pipx, as below, then even the version of build is not listed and the version of Python is unrelated.

File: .github/workflows/sdist.yml (hunk 1 above)
Comment (henryiii) [reply .github/workflows/sdist.yml:42]: pipx is a standard tool, it's part of the PyPA, just like build, setuptools, and pip. I can change this if preferred, but then you are dependent on `setup-python@v2`, while pipx is not.

**Comment by jreback** [comment]
@simonjayhawkins do we have any manual process in the release for sdists? i don't recall if this changed recently.

**Comment by jreback** [comment]
note completely averse here but this would need to address comments & rebase

**Comment by mroeschke** [comment]
Thanks for the PR, but it appears to have gone stale. If interested in continuing please merge the main branch, address the review, and we can reopen.

**Comment by henryiii** [comment]
I've rebased, but you can't reopen a PR after force pushing to it.  I think this was stuck because there was some "manual/other release process" that also needed updating but I wasn't pointed to that.

**Comment by mroeschke** [comment]
Ah okay. Yes I see we can't reopen this PR (we also don't recommend force pushing to branch as it makes the commit history and changes harder to view).

If you'd like to continue with a new PR we can try to contact the right person who has set up sdist.

**Comment by henryiii** [comment]
There weren't really any changes yet, it was just tracking the current main. I can retry.

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins do we have any manual process in the release for sdists? i don't recall if this changed recently.

yes, the sdist is created locally during the release process and then uploaded to the github release artifacts so that conda-forge can use it and also uploaded to PyPI

> Where are the SDist's built? https://github.com/MacPython/pandas-wheels only does wheels, AFAICT.

https://github.com/pandas-dev/pandas-release

and the conda recipe that points to the manually created sdist uploaded as a github release artifact is at 

https://github.com/conda-forge/pandas-feedstock

**Pull request closed without merging**